### PR TITLE
Changes runner logging to go through Logger instead of direct print statements

### PR DESCRIPTION
**Pull request by lourd**
Swaps out usage of `print` in the test runner for the appropriate `logger` methods, as brought up in #10696 

I took my best guess at what the correct levels were for each print, please let me know which ones you'd like to see changed. I also took out some `if DEBUG` or `if EMTEST_VERBOSE` checks where they became redundant when using `logger.debug`

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -892,7 +892,7 @@ def get_library(self, name, generated_libs, configure=['sh', './configure'],
     cache_name = ''.join([(c if c in valid_chars else '_') for c in cache_name])
 
     if self.library_cache.get(cache_name):
-      print('<load %s from cache> ' % cache_name, file=sys.stderr)
+      logger.debug('Loading %s from cache ' % cache_name)
```
Comment (kripken) [inline tests/runner.py]: this might be `info`

File: tests/runner.py [hunk 2]
```diff
@@ -901,7 +901,7 @@ def get_library(self, name, generated_libs, configure=['sh', './configure'],
         generated_libs.append(bc_file)
       return generated_libs
 
-    print('<building and saving %s into cache> ' % cache_name, file=sys.stderr)
+    logger.debug('Building and saving %s into cache ' % cache_name)
```
Comment (kripken) [inline tests/runner.py]: also `info`

File: tests/runner.py [hunk 3]
```diff
@@ -1354,38 +1350,49 @@ class BrowserCore(RunnerCore):
   def __init__(self, *args, **kwargs):
     super(BrowserCore, self).__init__(*args, **kwargs)
 
+  @staticmethod
+  def browser_open(url):
```
Comment (kripken) [inline tests/runner.py]: looks like this is the other PR's change? would be good to keep them in separate PRs.

Comment (kripken) [reply tests/runner.py]: oh, I missed the comment saying this can merge after that one, which resolves that issue. sounds good as is then.

File: tests/runner.py (hunk 1 above)
Comment (lourd) [reply tests/runner.py]: Done.

File: tests/runner.py (hunk 2 above)
Comment (lourd) [reply tests/runner.py]: Done.

**Comment by sbc100** [comment]
I'm not sure there is an advantage to making this change.

Just using print seems pretty reasonable in unit tests.   I agree that being able to hand all test output can be useful but I think that leaving that up to the test runner seems fine.

**Comment by lourd** [comment]
I was of the same mindset until I went through and made the changes. Doing so made clear that `print` wasn't sufficing, with the various checks on `DEBUG` and `EMTEST_VERBOSE`.  Defaulting to Logger also helps reduce the knowledge overhead required for correctly printing to `stderr` for errors.

If this change doesn't detract from the test runner behavior I'd say go ahead with it.

**Comment by sbc100** [comment]
OK.  Can you rebase though so we can see that change in isolation?

**Comment by lourd** [comment]
Sure, done.

File: tests/runner.py [hunk 4]
```diff
@@ -1693,13 +1686,13 @@ def build_library(name,
                              cwd=project_dir)
     except subprocess.CalledProcessError:
       with open(os.path.join(project_dir, 'configure_out')) as f:
-        print('-- configure stdout --')
-        print(f.read())
-        print('-- end configure stdout --')
+        logger.info('-- configure stdout --')
+        logger.info(f.read())
+        logger.info('-- end configure stdout --')
```
Comment (sbc100) [inline tests/runner.py]: I think I'd like to see how this kind of things looks in the output, before and after this change.   

In general I'm still feeling like this might not be worth the change.  But I'll check it out.

Comment (lourd) [reply tests/runner.py]: Alright, let me know if you need any changes from me.
